src/core/thread/mesh_forwarder.cpp

```diff
@@ -1047,7 +1047,9 @@ void MeshForwarder::HandleSentFrame(Mac::Frame &aFrame, otError aError)
 
             if (netif.GetMle().IsActiveRouter(neighbor->GetRloc16()))
             {
-                if (neighbor->GetLinkFailures() >= Mle::kFailedRouterTransmissions)
+                if ((netif.GetMle().GetRole() == OT_DEVICE_ROLE_ROUTER ||
+                     netif.GetMle().GetRole() == OT_DEVICE_ROLE_LEADER) &&
+                    neighbor->GetLinkFailures() >= Mle::kFailedRouterTransmissions)
```

Comment:
Just to confirm, if role is child, we want to go through the "child update" attempts instead of this (never allow this check)?

Reply:
On a second thought, this PR may not be necessary. will close it. Thanks.